### PR TITLE
Update providers

### PR DESCRIPTION
**Pull request by dkanada**
#40 had merge conflicts so I made a new one

### CODE REVIEW

File: Jellyfin.Plugin.Anime/Providers/AniSearch/AniSearchExternalId.cs [hunk 1]
```diff
@@ -7,23 +7,15 @@ namespace Jellyfin.Plugin.Anime.Providers.AniSearch
...
-        {
-            get { return ProviderNames.AniSearch; }
-        }
+            => ProviderNames.AniSearch;
 
         public string UrlFormatString
-        {
-            get { return "http://www.anisearch.de/anime/{0}"; }
-        }
+            => "http://www.anisearch.de/anime/{0}";
```
Comment (lenisko) [inline Jellyfin.Plugin.Anime/Providers/AniSearch/AniSearchExternalId.cs]: leftover?

Comment (dkanada) [reply Jellyfin.Plugin.Anime/Providers/AniSearch/AniSearchExternalId.cs]: Good catch!

File: Jellyfin.Plugin.Anime/Providers/GenreHelper.cs [hunk 2]
```diff
@@ -50,21 +50,6 @@ public static class GenreHelper
             {"Yaoi", "Adult"},
             {"Yuri", "Adult"},
```
Comment (heyhippari) [inline Jellyfin.Plugin.Anime/Providers/GenreHelper.cs:51]: Not really related to the changes in the PR, but I'm not a fan of this mapping.

To me, Yaoi or Yuri doesn't necessarily imply an "Adult" genre (i.e. it's not automatically porn). But it's also not a genre (Same with Shounen, Seinen and Josei, they're not genres but target audiences).

The "correct" way, imo, for this along with "genres" like Vampire, Witch, Shounen, Josei, Mecha, etc is to have the few general genres (Action, Adventure, Animation, Supernatural, etc) then the specifics as Tags.

So something like Mobile Suit Gundam would have genres as "Action, Science-Fiction, Animation" and tags like "Mecha, Space Opera".

File: Jellyfin.Plugin.Anime/Providers/GenreHelper.cs [hunk 3]
```diff
@@ -196,12 +181,6 @@ public static class GenreHelper
             "Vampire",
             "Youkai",
             "Zombie",
```
Comment (heyhippari) [inline Jellyfin.Plugin.Anime/Providers/GenreHelper.cs:183]: All these are from the German AniSearch (From L79 to L198)

File: Jellyfin.Plugin.Anime/Providers/GenreHelper.cs (hunk 2 above)
Comment (dkanada) [reply Jellyfin.Plugin.Anime/Providers/GenreHelper.cs:51]: I'd say this is out of scope for these changes.

Comment (heyhippari) [reply Jellyfin.Plugin.Anime/Providers/GenreHelper.cs:51]: For sure, as I said, it's more of a thought I had when reading the source :p

It seems to already work that way, but some of the mappings don't make sense. I'll whip something up when I have some time.

**Review by heyhippari — APPROVED** [review]

**Review by anthonylavado — APPROVED** [review]

File: Jellyfin.Plugin.Anime/Providers/AniDB/Metadata/AniDbSeriesProvider.cs [hunk 4]
```diff
@@ -53,8 +53,8 @@ public AniDbSeriesProvider(IApplicationPaths appPaths, IHttpClient httpClient)
             Current = this;
         }
 
-        internal static AniDbSeriesProvider Current { get; private set; }
-        public IAniDbTitleMatcher TitleMatcher { get; set; }
+        private static AniDbSeriesProvider Current { get; set; }
```
Comment (Bond-009) [inline Jellyfin.Plugin.Anime/Providers/AniDB/Metadata/AniDbSeriesProvider.cs:56]: Why don't we make these fields?

**Pull request merged**